### DISCUSSION TITLE
Boost mission writing by making a mission builder

### DISCUSSION THREAD

**Discussion by EmplOTM** [endless-sky/endless-sky#10866] [Ideas]
I've been playing this game for so many years and absolutely love it

I read many discussions about balance between Economy, Fight and Story game styles

In my experience the Story gameplay is really immersive, it provides awesome emotions and offers huge replay opportunities like any multi choice adventures.

I often build a fleet just for fun or to pass time, it is relaxing and fun

The economy part is fun and beings its satisfactions in the begining of a game but would benefit from a few added mechanics like investing or guilds that offer specific missions closing access to other guilds, trading aids sold in the outfitter to select automatically the best merchandise and memorizing good commercial routes

I often stop playing and put the game aside for periods of time because some of the mechanics are frustrating for me but always come back to it

I would absolutely love to write missions but the process is extremely tedious, I am convinced a lot of writers would contribute if writing missions did not involve so much of what creatives feel as "programming"

The best example I have in mind to explain my idea is map makers that could be found in old games, it made map making accessible to all

Another example are those programming for kids software that let you arrange blocks to make a "program"

Missions are built on Choices ( grab the gun or run away ) and conditions ( if you have the required ship and/or cargo then get to next step ) 

Putting effort in an interface ( an app or a website ) that would allow for writers to fill text fields relative to the story, have access to lists of conditions they could easily choose from ( drop down lists ? ) and concentrate on the story and the writing without having too much to do with the code would surely greatly improve the number of missions or stories submitted by the community and thus help balance the game naturally by offering lots of story arcs.

I am conscious it would be a lot of work an also convinced it would yield a lot of benefits and open the community to a lot of story-driven people

Any thoughts on this?

Tldr : build an app or website that would offer fields to fill and automatically generate the code for missions

**Comment by ziproot** [comment]
For some reason I cannot access the discussion, so I am 
trying to reply via email.

There’s this: https://github.com/shitwolfymakes/Endless-Sky-Mission-Builder/[1]

--------
[1] https://github.com/shitwolfymakes/Endless-Sky-Mission-Builder/
